oci/oci.go

```diff
@@ -1,27 +1,26 @@
-package daemon // import "github.com/docker/docker/daemon"
+package oci // import "github.com/docker/docker/oci"
 
 import (
 	"fmt"
 	"regexp"
 	"strconv"
 
-	"github.com/docker/docker/container"
-	"github.com/docker/docker/daemon/caps"
+	"github.com/docker/docker/oci/caps"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
 )
 
 // nolint: gosimple
-var (
-	deviceCgroupRuleRegex = regexp.MustCompile("^([acb]) ([0-9]+|\\*):([0-9]+|\\*) ([rwm]{1,3})$")
-)
+var deviceCgroupRuleRegex = regexp.MustCompile("^([acb]) ([0-9]+|\\*):([0-9]+|\\*) ([rwm]{1,3})$")
 
-func setCapabilities(s *specs.Spec, c *container.Container) error {
-	var caplist []string
-	var err error
-	if c.HostConfig.Privileged {
+func SetCapabilities(s *specs.Spec, add, drop []string, privileged bool) error {
```

Comment:
Wondering if we should extract the "diffing" out of this function, and make `SetCapabilities` take the actual list of capabilities instead (with the discussion on https://github.com/moby/moby/issues/25885 and https://github.com/moby/moby/pull/26849#issuecomment-249176129 in mind).

Perhaps it's orthogonal to this change, just that this will now be an exported function, so wondering if we should have the signature changed at the same time to match that design.


@justincormack ^^

Reply:
I'm not going to be doing refactoring of function changes.  I'm cleaning up imports and getting things in the right place for containerd integration.  Please review the changes and don't add more work ;)

Reply:
Yeah, that's fair. I can have a look at that in a follow up

Reply:
dont' mess with runtime code right now, i'm doing work.  Keep your hands out of it! ;)